### ISSUE TITLE
Use Charset for type-safety

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#454]
```
This also removes a source of UnsupportedEncodingException and thus 
IOException.  String.getBytes(Charset) requires Java 6:

http://docs.oracle.com/javase/6/docs/api/java/lang/String.html#getBytes(java.nio
.charset.Charset)

which H2 requires as per:

http://www.h2database.com/html/installation.html#requirements

I attached a patch.
```

Original issue reported on code.google.com by `g...@maginatics.com` on 16 Apr 2013 at 7:00

Attachments:
- [h2_charset.patch](https://storage.googleapis.com/google-code-attachments/h2database/issue-454/comment-0/h2_charset.patch)

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the patch! I will apply it, but it will take a few days.
```

Original comment by `thomas.t...@gmail.com` on 16 Apr 2013 at 8:38
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This is now committed to the trunk.
```

Original comment by `thomas.t...@gmail.com` on 21 Apr 2013 at 6:11

**Comment by GoogleCodeExporter** [comment]
```
Fixed in version 1.3.172.
```

Original comment by `thomas.t...@gmail.com` on 25 May 2013 at 1:48
- Changed state: **Fixed**
